### PR TITLE
fix(CORE/dbupdate)): allow to pass MySQL password via env

### PR DESCRIPTION
**Pull request by Yehonal**
## Changes Proposed:

- allow to pass MySQL password via env. This will stop the mysql password warning issue 
- fixed devContainer override file issue
- added authserver dry-run in our CI

## Issues Addressed:

- Closes 

## Tests Performed:

This:

image

Instead of this:

image

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/common/Utilities/StartProcess.cpp [hunk 1]
```diff
@@ -93,11 +93,13 @@ namespace Acore
         {
             if (inputFile)
             {
+
+
```
Comment (acidmanifesto) [inline src/common/Utilities/StartProcess.cpp:97]: white space

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

**Comment by Winfidonarleyan** [comment]
This warnings only for mysql

**Review by acidmanifesto — APPROVED** [review]

**Comment by Yehonal** [comment]
> This warnings only for mysql

Can you test this PR on another db?

**Comment by Naios** [comment]
@Yehonal Seems like you have fixed a long outstanding issue of my DBUpdater implementation, nice.
We would like to cherry-pick this fix to TrinityCore as well.

Could you agree to license your change also under GPL? Otherwise we cannot pick it over because AGPL (AzerothCore) is more constrained than GPL (TrinityCore).

**Comment by Yehonal** [comment]
> @Yehonal Seems like you have fixed a long outstanding issue of my DBUpdater implementation, nice. We would like to cherry-pick this fix to TrinityCore as well.
> 
> Could you agree to license your change also under GPL? Otherwise we cannot pick it over because AGPL (AzerothCore) is more constrained than GPL (TrinityCore).

No prob, as long as you set `-c user.name=Yehonal-c user.email=yehonal.azeroth@gmail.com`  while cherry-picking

**Comment by Nyeriah** [comment]
It is worth mentioning that AGPLv3 disregards user consent when it comes to the propagation of the license. That means a file receiving code originally licensed as AGPLv3 will necessarily follow the more restrictive license.

**Comment by Naios** [comment]
@Yehonal thank you, of course the commit will be made with you as original author listed.

@Nyeriah You can never disregard user consent if he allows that his own intellectual work is also licensed under a another license.
By following your logic Yehonal would waive their rights to his original work by just commiting it to an AGPL repo, which is not the case. You always keep ownership of your own (personal) work regardless if you decide to sublicense it through an open-source licensed or not. Therefore Yehonal can allow to sublicense their changes under the GPL as well. *Note: this is no legal advice.*

**Comment by Nyeriah** [comment]
@Naios it’s the non-legal advice we got from FSF

**Comment by Yehonal** [comment]
New PR available here: https://github.com/azerothcore/azerothcore-wotlk/pull/13449

**Pull request merged**